deploy.py

```diff
@@ -447,7 +447,11 @@ def provision_execute(name):
     for stage in stages:
         logging.info(f"Running stage")
         results = []
-        with concurrent.futures.ThreadPoolExecutor() as executor:
+        max_executors = len(hosts)
+        # The last stage should be synchronous
```

Comment:
as @arbulu89  commented in a previous comment we should add here more context to this design comment.

like

` # The last stage should be synchronous because XX blabla` with context

Reply:
Yes, agree. Otherwise we might not understand in the future why we did this

Reply:
Done.